File: Array/TypeScript/_0976_LargestPerimeterTriangle.ts
/**
 * Iteration
 * Time Complexity: BigO(N)
 * Space Complexity: BigO(1)
 */
function largestPerimeter(nums: number[]): number {
  let desc = nums.sort((a, b) => b - a);
  for (let i = 0; i < desc.length - 2; i++) {
    if (desc[i] < desc[i + 1] + desc[i + 2]) {
      return desc[i] + desc[i + 1] + desc[i + 2];
    }
  }
  return 0;
}

console.log(largestPerimeter([2, 1, 2]));
console.log(largestPerimeter([1, 2, 1, 10]));

File: Array/Python/_0976_LargestPerimeterTriangle.py
from typing import List


class Solution:
    # Iteration
    # Time Complexity: BigO(N)
    # Space Complexity: BigO(1)
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort(reverse=True)
        for i in range(len(nums) - 2):
            if nums[i] < nums[i+1] + nums[i+2]:
                return nums[i] + nums[i+1] + nums[i+2]
        return 0


ins = Solution()

print(ins.largestPerimeter([2, 1, 2]))
print(ins.largestPerimeter([1, 2, 1, 10]))
